src/main/java/org/eclipse/cargotracker/domain/model/cargo/Cargo.java

```diff
@@ -168,7 +168,7 @@ public boolean equals(Object object) {
     if (this == object) {
       return true;
     }
-    if (object == null || getClass() != object.getClass()) {
+    if (object == null || !(object instanceof Cargo)) {
```

Comment:
Let's decouple these into a separate PR in the master branch to reduce clutter here and focus on the actual portability changes.

Reply:
If this is moved to a separate PR, this could be merged firstly, in Hibernate entity comparation depends on this.